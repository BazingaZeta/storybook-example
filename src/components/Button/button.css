.button {
  background: #2d35a7;
  border: none;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 16px;
  margin: 0 auto 24px;
  padding: 12px 24px;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.5);
}

.button:disabled {
  background: lightgray;
  cursor: not-allowed;
}

.button:active:not(:disabled) {
  left: 1px;
  position: relative;
  top: 1px;
}

.button--primary {
  background-color: #1ea7fd;
}

.button--secondary {
  box-shadow: none;
  color: lightgray;
}

.button--small {
  font-size: 12px;
  padding: 10px 16px;
}

.button--medium {
  font-size: 14px;
  padding: 11px 20px;
}

.button--large {
  font-size: 16px;
  padding: 12px 24px;
}
