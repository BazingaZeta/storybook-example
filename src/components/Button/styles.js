import styled from 'styled-components'

const SIZES = {
  small: '0.85rem',
  medium: '1rem',
  large: '1.15rem'
}

const bgColor = ({ primary, backgroundColor }) =>
  backgroundColor || (primary ? '#1ea7fd' : '#2d35a7')

export const Button = styled.button`
  background-color: ${props => bgColor(props)};
  border: none;
  border-radius: 0.5rem;
  color: ${props => (props.primary ? '#fff' : 'lightgray')};
  box-shadow: ${props => (props.primary ? 'none' : 'inherit')};
  cursor: pointer;
  display: block;
  font-size: ${props => SIZES[props.size]};
  margin: 0 auto 1rem;
  padding: 1rem 2rem;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.5);

  &:disabled {
    background-color: lightgray;
    color: #fff;
    cursor: not-allowed;
  }

  &:active:not(:disabled) {
    left: 1px;
    position: relative;
    top: 1px;
  }
`
