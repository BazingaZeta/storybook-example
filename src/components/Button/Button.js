import React from 'react'
import PropTypes from 'prop-types'
import { Button as StyledButton } from './styles'

const Button = ({ disabled, primary, backgroundColor, size, label, ...props }) => (
  <StyledButton
    type="button"
    disabled={disabled}
    primary={primary}
    size={size}
    backgroundColor={backgroundColor}
    {...props}>
    {label}
  </StyledButton>
)

Button.propTypes = {
  disabled: PropTypes.bool,
  primary: PropTypes.bool,
  backgroundColor: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

Button.defaultProps = {
  disabled: false,
  primary: false,
  backgroundColor: null,
  size: 'medium'
}

export default Button
