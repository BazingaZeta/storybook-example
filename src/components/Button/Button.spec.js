import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Button from '.'

describe('components > Button', () => {
  const onClick = jest.fn()
  const label = 'plus me up!'

  let component

  describe.each`
    primary  | size        | disabled
    ${true}  | ${'small'}  | ${true}
    ${true}  | ${'small'}  | ${false}
    ${true}  | ${'medium'} | ${true}
    ${true}  | ${'medium'} | ${false}
    ${true}  | ${'large'}  | ${true}
    ${true}  | ${'large'}  | ${false}
    ${false} | ${'small'}  | ${true}
    ${false} | ${'small'}  | ${false}
    ${false} | ${'medium'} | ${true}
    ${false} | ${'medium'} | ${false}
    ${false} | ${'large'}  | ${true}
    ${false} | ${'large'}  | ${false}
  `('renders', ({ primary, size, disabled }) => {
    beforeAll(() => {
      component = render(
        <Button primary={primary} size={size} disabled={disabled} onClick={onClick} label={label} />
      )
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })

  describe('clicking on it', () => {
    let button

    beforeAll(() => {
      component = render(<Button onClick={onClick} label={label} />)
      button = component.getByRole('button')
      fireEvent.click(button)
    })

    it('passed click on to onClick', () => {
      expect(onClick).toHaveBeenCalledTimes(1)
    })
  })
})
