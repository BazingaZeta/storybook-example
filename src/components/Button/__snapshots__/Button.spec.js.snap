// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Button renders rendered without crashing 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx hAyApL"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 2`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx hAyApL"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 3`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx iVLmaZ"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 4`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx iVLmaZ"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 5`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx lolljv"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 6`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx lolljv"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 7`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx iwTwVc"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 8`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx iwTwVc"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 9`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx fUwXgK"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 10`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx fUwXgK"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 11`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx dpAZAU"
    disabled=""
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;

exports[`components > Button renders rendered without crashing 12`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx dpAZAU"
    type="button"
  >
    plus me up!
  </button>
</DocumentFragment>
`;
